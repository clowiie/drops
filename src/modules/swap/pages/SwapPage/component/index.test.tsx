import React from 'react'

import { render } from '@testing-library/react'

import mockComponent from 'test/utils/mockComponent'

import * as SwapPageType from '.'

describe('SwapPage', () => {
  const SwapPlateSpy = mockComponent('modules/swap/components/SwapPlate')

  const { default: SwapPage } = require('.') as typeof SwapPageType

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should render swap page', () => {
    const { container, getByText } = render(<SwapPage />)

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText('Swap')).toBeDefined()
    expect(SwapPlateSpy).toBeCalledTimes(1)
  })
})
