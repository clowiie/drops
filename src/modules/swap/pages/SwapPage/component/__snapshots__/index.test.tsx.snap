// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapPage should render swap page 1`] = `
<div
  class="flex flex-col justify-center items-center pt-14"
>
  <div
    class="mb-8"
  >
    <span
      class="text-2xl font-semibold text-center"
    >
      Swap
    </span>
  </div>
  <div />
</div>
`;
